test/api/annotations.cpp

```diff
@@ -79,6 +79,24 @@ TEST(Annotations, FillAnnotation) {
     test.checkRendering("fill_annotation_max_zoom");
 }
 
+TEST(Annotations, OverlappingFillAnnotation) {
+    AnnotationTest test;
+
+    Polygon<double> polygon = {{ {{ { 0, 0 }, { 0, 45 }, { 45, 45 }, { 45, 0 } }} }};
+    FillAnnotation underlaidAnnotation { polygon };
+    underlaidAnnotation.color = { { 0, 255, 0, 1 } };
+    FillAnnotation overlaidAnnotation { polygon };
+    overlaidAnnotation.color = { { 255, 0, 0, 1 } };
+
+    test.map.setStyleJSON(util::read_file("test/fixtures/api/empty.json"));
+    test.map.addAnnotation(underlaidAnnotation);
+    test.map.addAnnotation(overlaidAnnotation);
+    test.checkRendering("fill_annotation");
+
+    test.map.setZoom(test.map.getMaxZoom());
```

Comment:
I think we can remove this block. It's a good check for the basic fill annotation test above but not necessary here.